de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/csv/CsvFileReader.java

```diff
@@ -126,29 +85,35 @@ public boolean closeFile() {
 	 * @param startLine the first line number to read
 	 * @param endLine the last line number to read
 	 * @return the CSV content as matrix of two lists
+	 * @throws IOException 
 	 */
-	public List<List<String>> readCsvFile(int startLine, int endLine) {
+	public List<List<String>> readCsvFile(String filePath, int startLine, int endLine) throws IOException {
 		List<List<String>> csvContentMatrix = new ArrayList<List<String>>();
-		
-		if (records == null) {
-			return csvContentMatrix;
-		}
-		
-		int lineNumber = 0;
-		for (CSVRecord record : records) {
-			if (lineNumber >= startLine) {
-				List<String> reqData = new ArrayList<String>();
-				for (String att : record) {
-					reqData.add(att);
+		Path csvFilePath = Paths.get(filePath);
+
+		FileReader fr = new FileReader(csvFilePath.toFile());
+		try {

```

Comment:
I would strongly recomment using try-with-resources here: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

It would look like this:

```
try (FileReader fr = new FileReader(csvFilePath.toFile())) {
...
}
```

It takes care of closing the resource both in case of a failure and a success.